js/js.dce/src/org/jetbrains/kotlin/js/dce/ReachabilityTracker.kt

```diff
@@ -134,23 +138,33 @@ class ReachabilityTracker(
 
     override fun visitContinue(x: JsContinue) { }
 
+    private fun collectReachable(node: Node) {
+        if (!collectOnlyRootNodes) {
+            reachableNodesImpl += node
+        } else {
+            node.extractRoot(visitedTag)?.let {
+                reachableNodesImpl += it
+            }
+        }
+    }
+
     fun reach(node: Node) {
         if (node.reachable) return
         node.reachable = true
-        reachableNodesImpl += node
+        collectReachable(node)
 
         reachDeclaration(node)
 
         reachDependencies(node)
         node.members.toList().forEach { (name, member) ->
             if (!member.reachable) {
-                reportAndNest("reach: member $name", null) { reach(member) }
+                nested { reach(member) }
```

Comment:
Is it OK to simply remove the logging?

What exactly is slow? Message construction? Couldn't we achieve the same by calculating the message in a lambda?

Reply:
Brought back logging when `printReachabilityInfo` argument is set to `true`